sherpa/optmethods/tests/test_optmethods.py

```diff
@@ -39,13 +38,22 @@ def tst_all( self, name, fct, fmin, x0, xmin, xmax,
                  iprint=False ):
         self.tst( optfcts.neldermead, name + self.nm, fct, fmin,
                   x0, xmin, xmax, iprint=iprint )
-##         self.tst( stogo.stogo, name + '_stogo', fct, fmin,
-##                   x0, xmin, xmax, iprint=iprint )
         self.tst( optfcts.montecarlo,  name + self.mc , fct, fmin,
                   x0, xmin, xmax, iprint=iprint, maxfev=8192*len(x0) )
         self.tst( optfcts.lmdif, name + self.lm, fct, fmin,
                   x0, xmin, xmax, iprint=iprint )
 
+        #
+        # If the following lines are uncommented out, we get errors
+        # when running the optimization test
+        #
+        # try:
+        #     import iminuit
+        #     self.tst(optfcts.midnight, name + '_minuit', fct, fmin,
+        #              x0, xmin, xmax, iprint=iprint)
+        # except ImportError:
+        #     pass
+
```

Comment:
Made the __all__ += changes, but get the following error when I tried to
push:

Password for 'https://dtnguyen2@github.com':
To https://github.com/sherpa/sherpa
 ! [rejected]        feature/minuit -> feature/minuit (non-fast-forward)
error: failed to push some refs to 'https://github.com/sherpa/sherpa'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.

What's a fella to do?

On Fri, Mar 25, 2016 at 8:37 AM, Doug Burke <notifications@github.com>
wrote:

> In sherpa/optmethods/tests/test_optmethods.py
> <https://github.com/sherpa/sherpa/pull/194#discussion_r57440534>:
>
> >          self.tst( optfcts.montecarlo,  name + self.mc , fct, fmin,
> >                    x0, xmin, xmax, iprint=iprint, maxfev=8192*len(x0) )
> >          self.tst( optfcts.lmdif, name + self.lm, fct, fmin,
> >                    x0, xmin, xmax, iprint=iprint )
> >
> > +        #
> > +        # If the following lines are uncommented out, we get errors
> > +        # when running the optimization test
> > +        #
> > +        # try:
> > +        #     import iminuit
> > +        #     self.tst(optfcts.midnight, name + '_minuit', fct, fmin,
> > +        #              x0, xmin, xmax, iprint=iprint)
> > +        # except ImportError:
> > +        #     pass
> > +
>
> Rather than protecting the test in a try/except block, we should use the
> unittest.skipIf decorator (or one of the wrappers around such as
> sherpa.utils.requires_package), since then we have better tracking of the
> total number of tests. @olaurino <https://github.com/olaurino> can
> (hopefully) provide info on the preferred syntax here.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/sherpa/sherpa/pull/194/files/3d8653dfe45d619b1a8a514d7dda345dd38b655f#r57440534>
>